src/legacy/core_plugins/vis_default_editor/public/components/agg_params.tsx

```diff
@@ -83,18 +84,24 @@ function DefaultEditorAggParams({
   allowedAggs = [],
   hideCustomLabel = false,
 }: DefaultEditorAggParamsProps) {
-  const schema = getSchemaByName(schemas, agg.schema);
-  const { title } = schema;
-  const aggFilter = [...allowedAggs, ...(schema.aggFilter || [])];
```

Comment:
Quick question, why did you migrate to the usage of `useMemo`, don't seem to be the results of expensive operations to memorize?

Reply:
It is important because of referential equality.
On every render the `aggFilter` would be a completely new array, which would cause recalculation of `groupedAggTypeOptions` (if if nothing were changed), which would return a completely new array of agg type options in its turn, and would make to rerender the `DefaultEditorAggSelect` component.
So using memoization in react is not only for expensive computations, but also for preventing additional re-renders or component tree.

Reply:
👍 thanks for explanation!